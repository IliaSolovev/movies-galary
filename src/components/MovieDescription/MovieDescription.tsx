import React from 'react';
import s from './movieDescription.module.scss';
import { Movie } from '../../redux/moviesSlice';

interface Props {
    data: Movie
}

const MovieDescription: React.FC<Props> = ({ data, ...rest }) => (
  <div className={s.content}>
    {data === null && <p>Loading...</p>}
    {data
            && (
            <>
              <div className={s.poster}>
                <img src={data.poster_path} alt={data.title} />
              </div>
              <div className={s.description}>
                <h3 className={s.description__title}>
                  {data.title}
                  {' '}
                  <div className={s.description__rating}>{data.vote_average}</div>
                </h3>
                <h4 className={s.description__subtitle}>{data.tagline}</h4>
                <div className={s.description__durationRelease}>
                  <div className={s.description__release}>{data.release_date.split('-')[0]}</div>
                  <div className={s.description__duration}>
                    {data.runtime}
                    {' '}
                    min
                  </div>
                </div>
                <div className={s.description__overview}>{data.overview}</div>
              </div>
            </>
            )}

  </div>
);

export default MovieDescription;
