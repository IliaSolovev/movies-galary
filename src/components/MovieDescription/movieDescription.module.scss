@import '../../styles/variables';
@import '../../styles/main';

.content{
  @extend .container;
  display: grid;
  grid-template-columns: 1fr 3fr;
  grid-gap: 5vw;
  @media screen and (max-width: 1000px){
    grid-template-columns: minmax(250px,350px) minmax(230px,350px);
  }
  @media screen and (max-width: 750px){
    margin: 0 auto;
    grid-template-rows: min-content ;
    grid-template-columns: minmax(200px,300px) minmax(230px,350px);
  }
  @media screen and (max-width: 580px){
    margin: 0 auto;
    grid-template-columns: 1fr;
    grid-template-rows: minmax(200px,350px) min-content;
  }

}
.poster{
  display: flex;
  justify-content: center;
  & > img {
    width: 100%;
    height: 100%;
    @media screen and (max-width: 750px){
      width: 240px;
      height: 340px;
    }
    @media screen and (max-width: 580px){
      max-width: 300px;
      max-height: 400px;
    }
    @media screen and (max-width: 400px){
      max-width: 280px;
      max-height: 350px;
    }
  }
}

.description{
  font-family: $mainFont;
  &__titles{

  }
  &__title{
    margin: 25px 0 5px 0;
    font-size: 1.7em;
    font-weight: bold;
    color: $redColor;
  }
  &__subtitle{
    margin:  0 0 10px 0;
    font-size: 1.1em;
    font-weight: lighter;
    color: $greyColor;
  }
  &__rating{
    display: inline-block;
    position: absolute;
    margin-left: 5px;
    width: 40px;
    height: 40px;
    padding: 9px 0 0 0 ;
    font-size: .7em;
    font-weight: 300;
    text-align: center;
    color:$greenColor;
    border: 1px solid $greenColor;
    border-radius: 50%;
    box-sizing: border-box;
  }

  &__durationRelease{
    width: 135px;
    display: flex;
    justify-content: space-between;
    margin-bottom: 25px;
    font-size: 1.1em;
    font-weight: normal;
    color: $greyColor;
  }
  &__release{

  }
  &__duration{

  }
  &__overview{
    margin-bottom: 10px;
    font-size: 1.2em;
    font-weight: normal;
    color: $greyColor;
  }
}
