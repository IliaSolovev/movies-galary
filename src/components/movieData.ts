import { MovieCardData, MovieDescriptionData } from '../types';

export const movieCard2: MovieCardData = {
  id: '337167',
  title: 'Fifty Shades Freed',
  release_date: '2018-02-07',
  poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
  genres: [
    'Drama',
    'Romance',
  ],
};
export const movieCard1: MovieCardData = {
  id: '337167',
  title: 'Fifty Shades Freed',
  release_date: '2019-02-07',
  poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
  genres: [
    'Drama',
    'Romance',
  ],
};

export const movieDescription: MovieDescriptionData = {
  id: '337167',
  title: 'Fifty Shades Freed',
  tagline: "Don't miss the climax",
  rating: 6.1,
  release_date: '2019-02-07',
  poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
  overview: 'Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable '
    + 'connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, '
    + 'new threats could jeopardize their happy ending before it even begins.',
  genres: [
    'Drama',
    'Romance',
  ],
  runtime: 106,
};
