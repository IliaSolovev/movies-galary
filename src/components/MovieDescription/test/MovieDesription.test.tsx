import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MovieDescription } from '../MovieDescription';
import { movieDescription } from '../../movieData';


it('should render movie description', () => {
  const component = shallow(<MovieDescription movie={movieDescription} />);
  expect(toJson(component)).toMatchSnapshot();
});
