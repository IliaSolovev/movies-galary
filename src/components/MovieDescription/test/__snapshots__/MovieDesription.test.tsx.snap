// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render movie description 1`] = `
<div
  className="content"
>
  <div
    className="poster"
  >
    <img
      alt="Fifty Shades Freed"
      src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
    />
  </div>
  <div
    className="description"
  >
    <h3
      className="description__title"
    >
      Fifty Shades Freed
       
      <div
        className="description__rating"
      >
        6.1
      </div>
    </h3>
    <h4
      className="description__subtitle"
    >
      Don't miss the climax
    </h4>
    <div
      className="description__durationRelease"
    >
      <div>
        2018
      </div>
      <div>
        106
         
        min
      </div>
    </div>
    <div
      className="description__overview"
    >
      Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
    </div>
  </div>
</div>
`;
